src/python/utils/index_storage.py

```diff
@@ -27,6 +28,22 @@ def get_partition(self, partition_uri):
                    .format(partition_uri))
             raise ex
 
+    def get_s3_keys(self):
```

Comment:
I know I initially told you to move this to the index storage class, but that's because I for some reason thought the keys were coming from S3. Could you move this to the index_controller class so that all interactions with the index_metadata_table are consolidated in one place? Also can you rename it to something more implementation agnostic such as get_partition_ids()?